src/OpenTelemetry.Instrumentation.EventCounters/EventCounterListener.cs

```diff
@@ -0,0 +1,245 @@
+﻿// <copyright file="EventCounterListener.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Metrics;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation.EventCounters
+{
+    internal class EventCounterListener : EventListener
+    {
+        // EventSourceCreated can run before constructor, so constructor injection of the options dependency will not work.
+        // Using static to set options at host build time.
+        public static EventCounterMetricsOptions EventCounterMetricsOptions = new EventCounterMetricsOptions();
```

Comment:
its a bit odd. But I think we can fix this later.

Reply:
Ill fix this, we can wait until initialization to read options.

Reply:
I made this change to add list of sources to a concurrent bag,  then enable events after initialization with options from constructor.